n3fit/src/n3fit/model_trainer.py

```diff
@@ -795,14 +796,17 @@ def hyperparametrizable(self, params):
         l_valid = []
         l_exper = []
         l_hyper = []
+        # And lists to save hyperopt utilities
+        n3pdfs = []
+        exp_models = []
 
         ### Training loop
         for k, partition in enumerate(self.kpartitions):
             # Each partition of the kfolding needs to have its own separate model
             # and the seed needs to be updated accordingly
             seeds = self._nn_seeds
             if k > 0:
-                seeds = [np.random.randint(0, pow(2, 31))]
+                seeds = [np.random.randint(0, pow(2, 31)) for _ in seeds]
```

Comment:
Maybe this is completely irrelevant for hyperopt, but you can for example get different seeds for replica 2 depending on wheter you ran `n3fit 2` or `n3fit 1 -r 2`.

Reply:
When doing hyperopt these are two different things so it's fine (the first command will populate only `replica_2` while the second populates only `replica_1`)